chitchat/src/state.rs

```diff
@@ -1,21 +1,23 @@
 use std::collections::btree_map::Entry;
-use std::collections::{BTreeMap, BinaryHeap, HashSet};
+use std::collections::{BTreeMap, HashSet};
 use std::net::SocketAddr;
 use std::time::Instant;
 
+use itertools::Itertools;
 use rand::prelude::SliceRandom;
 use rand::Rng;
 use serde::{Deserialize, Serialize};
 use tokio::sync::watch;
 
 use crate::delta::{Delta, DeltaWriter};
-use crate::digest::Digest;
-use crate::{ChitchatId, Version, VersionedValue};
+use crate::digest::{Digest, NodeDigest};
+use crate::{ChitchatId, Heartbeat, MaxVersion, Version, VersionedValue};
 
 #[derive(Debug, Clone, Serialize, Deserialize)]
 pub struct NodeState {
+    pub(crate) heartbeat: Heartbeat,
     pub(crate) key_values: BTreeMap<String, VersionedValue>,
-    pub(crate) max_version: u64,
+    pub(crate) max_version: MaxVersion,
```

Comment:
Remark that you can ignore, I think this is a discussion we can have if we want to write some coding style guidelines.

I'm not convinced by using a type alias to replace a simple type such as u64. I don't see the added value and generally, I found the code easier to understand when I see the real type `u64`.